ccm_web/server/src/api/api.service.ts

```diff
@@ -25,6 +25,29 @@ export class APIService {
     }
   }
 
+  async getCourseSections(userLoginId: string, courseId: number): Promise<CanvasCourseSection[] | APIErrorData> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId, '/api/v1/')
+    try {
+      const endpoint = `courses/${courseId}/sections`
+      const queryParams = { 'include': ['total_students'] } // use list for "include" values
+      logger.debug(`Sending request to Canvas (get all pages) - Endpoint: ${endpoint}; Method: GET`)
+      // FIXME: list() should return promise, toArray() should be callable later
+      const sectionsFull = await requestor.list<CanvasCourseSection>(endpoint, queryParams).toArray()
+      // FIXME: no access to got Response; statusCode, etc. not available!
+      // logger.debug(`Received response with status code ${sectionsFull.statusCode}`) // broken
+      const sections = sectionsFull.map(s => ({
```

Comment:
Although my TypeScript experience is limited, it seems that the declaration of `list()` and its `toArray()` method could be better.

The way this Canvas API function was called before, without pagination handling via `get()`, the got `Response` object that was returned had a `statusCode` property.  It seems that this `list()` method doesn't return that type of object, so that property (and several others that would be useful) isn't available.

Also, it seems that the `toArray()` call here is required, because that is actually returning the promise.  It seems to me that `list()` should return a promise for a got `Response` object **_and_** the `toArray()` method should be callable at any time on that object.  That is, maybe in some case the code may want to use the `Response` object for some operations, then its `toArray()` method could be called later to get the response contents as an array.

For example, I'd expect it to be used this way…

```suggestion
      // FIXME: list() should return promise, toArray() should be callable later
      const response = await requestor.list<CanvasCourseSection>(endpoint, queryParams)
      // FIXME: no access to got Response; statusCode, etc. not available!
      logger.debug(`Received response with status code ${response.statusCode}`)
      const sections = response.toArray().map(s => ({
```

I will open a new issue for this.

Reply:
Opened issue #159.